### PR TITLE
Bugfix/group reserved kw union data v3

### PR DESCRIPTION
**Pull request by fivetran-jamie**
## PR Overview
**This PR will address the following Issue/Feature:**
https://getdbt.slack.com/archives/C01D1R2JLLA/p1733513338340409

**This PR will result in the following new package version:**

v0.14.1

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fixes
- In v0.19.0, Snowflake users may have seen `when searching for a relation, dbt found an approximate match` errors when running the `stg_zendesk__group_tmp` model. We have updated the `adapter.get_relation()` logic that was causing this error.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [ ] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Recreated the issue in snowflake by creating a `group` table (selecting from the `group_data` seed file) 
```
  When searching for a relation, dbt found an approximate match. Instead of guessing 
  which relation to use, dbt will move on. Please delete "DBT_PACKAGE_TESTING"."ZZ_DBT_JAMIE_1"."GROUP", or rename it to be less ambiguous.
  Searched for: DBT_PACKAGE_TESTING.ZZ_DBT_JAMIE_1."GROUP"
  Found: "DBT_PACKAGE_TESTING"."ZZ_DBT_JAMIE_1"."GROUP"
  
  > in macro default__union_zendesk_connections (macros/union/union_zendesk_connections.sql)
  > called by macro union_zendesk_connections (macros/union/union_zendesk_connections.sql)
  > called by model stg_zendesk__group_tmp (models/tmp/stg_zendesk__group_tmp.sql)
```

After applying the changes, confirmed that it works in Snowflake, BigQuery, and Redshift (@fivetran-joemarkiewicz will run on Postgres as I am having some terminal issues) when the table is named `group`, when the identifier var is used (ie to run on `group_data`), and when the table is missing (I changed the identifier var value to something random like `group_fake_alias`)

### If you had to summarize this PR in an emoji, which would it be?
 
🏂

### CODE REVIEW

File: macros/union/union_zendesk_connections.sql [hunk 1]
```diff
@@ -42,12 +42,18 @@
...
     {%- set relation=adapter.get_relation(
         database=source(single_source_name, single_table_name).database,
         schema=source(single_source_name, single_table_name).schema,
-        identifier=source(single_source_name, single_table_name).identifier
+        identifier=var(identifier_var, single_table_name)
     ) -%}
 
+    {# {{ log('Old identifier = ' ~ source(single_source_name, single_table_name).identifier, info=true) }}
+    {{ log('New identifier = ' ~ identifier_var, info=true) }} #}
+
```
Comment (fivetran-joemarkiewicz) [inline macros/union/union_zendesk_connections.sql:56]: We should remove the logs when merging and release.
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+# dbt_zendesk_source v0.14.1
+
+[PR #58](https://github.com/fivetran/dbt_zendesk_source/pull/58) includes the following update:
+
+## Bug Fixes
+- In v0.14.0 (or [v0.19.0](https://github.com/fivetran/dbt_zendesk/releases/tag/v0.19.0) of the transform package), Snowflake users may have seen `when searching for a relation, dbt found an approximate match` errors when running the `stg_zendesk__group_tmp` model. We have updated the `adapter.get_relation()` logic that was causing this error.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: Suggestion for just a bit clearer CHANGELOG entry. 

```suggestion
- In v0.14.0 (or [v0.19.0](https://github.com/fivetran/dbt_zendesk/releases/tag/v0.19.0) of the transform package), Snowflake users may have seen `when searching for a relation, dbt found an approximate match` errors when running the `stg_zendesk__group_tmp` model. The issue stemmed from the `adapter.get_relation()` logic within the `union_zendesk_connections` macro, which has now been updated to resolve the error.
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie I was able to confirm that this is not/no longer an issue on Postgres when running locally with the `group` table retaining the reserved word naming convention.

I just have a request to make sure you remove the commented out Jinja logs are removed as well as a small wording suggestion update in the CHANGELOG. Otherwise this looks good to go!

**Review by fivetran-catfritz — APPROVED** [review]
lgtm!

**Pull request merged**